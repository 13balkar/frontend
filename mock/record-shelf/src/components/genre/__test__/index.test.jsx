import React from 'react';
import { render } from '@testing-library/react';
import Genre from '..';

const mockPost = [
  {
    id: 'cd3cc1e3-e1e0-4ccd-bc67-747648985838',
    name: 'Lost',
    imageUrl: 'https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631',
    artist: {
      id: '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
      name: 'Maroon 5'
    },
    genre: {
      id: '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
      name: 'Pop'
    }
  }
];

describe('Genre', () => {
  it('should render Genre correctly', () => {
    const { asFragment } = render(<Genre post={mockPost} genre='Pop'/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
