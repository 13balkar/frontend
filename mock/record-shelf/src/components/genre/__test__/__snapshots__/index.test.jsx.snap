// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre should render Genre correctly 1`] = `
<DocumentFragment>
  <div
    class="genre"
  >
    <div
      class="heading"
    >
      <img
        alt="Pop"
        src="/assets/genre-Pop.png"
      />
      <div
        class="name"
      >
        <h1>
          Pop
        </h1>
      </div>
    </div>
    <div
      class="cardList"
    >
      <div
        class="card"
      >
        <img
          alt="img"
          class="img"
          src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
        />
        <div
          class="text"
        >
          <div
            class="card-desc"
          >
            <p
              class="song"
            >
              Lost
            </p>
            <p
              class="singer"
            >
              Maroon 5
            </p>
          </div>
          <div>
            loading
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
